lib/ecmascript.ts

```diff
@@ -3849,397 +4038,366 @@ export const ES = ObjectAssign({}, ES2020, {
+      const rounded = RoundNumberToIncrement(nanoseconds, divisor.multiply(increment).toJSNumber(), roundingMode);
+      total = nanoseconds.toJSNumber() / divisor.toJSNumber();
+      days = rounded.divide(divisor).toJSNumber();
+      nanoseconds = 0;
+      break;
+    }
+    case 'hour': {
+      const divisor = 3600e9;
+      nanoseconds = bigInt(hours)
+        .multiply(3600e9)
+        .plus(bigInt(minutes).multiply(60e9))
+        .plus(bigInt(seconds).multiply(1e9))
+        .plus(bigInt(milliseconds).multiply(1e6))
+        .plus(bigInt(microseconds).multiply(1e3))
+        .plus(nanoseconds);
+      total = nanoseconds.toJSNumber() / divisor;
+      const rounded = RoundNumberToIncrement(nanoseconds, divisor * increment, roundingMode);
+      hours = rounded.divide(divisor).toJSNumber();
+      minutes = seconds = milliseconds = microseconds = nanoseconds = 0;
+      break;
+    }
+    case 'minute': {
+      const divisor = 60e9;
+      nanoseconds = bigInt(minutes)
+        .multiply(60e9)
+        .plus(bigInt(seconds).multiply(1e9))
+        .plus(bigInt(milliseconds).multiply(1e6))
+        .plus(bigInt(microseconds).multiply(1e3))
+        .plus(nanoseconds);
+      total = nanoseconds.toJSNumber() / divisor;
+      const rounded = RoundNumberToIncrement(nanoseconds, divisor * increment, roundingMode);
+      minutes = rounded.divide(divisor).toJSNumber();
+      seconds = milliseconds = microseconds = nanoseconds = 0;
+      break;
+    }
+    case 'second': {
+      const divisor = 1e9;
+      nanoseconds = bigInt(seconds)
+        .multiply(1e9)
+        .plus(bigInt(milliseconds).multiply(1e6))
+        .plus(bigInt(microseconds).multiply(1e3))
+        .plus(nanoseconds);
+      total = nanoseconds.toJSNumber() / divisor;
+      const rounded = RoundNumberToIncrement(nanoseconds, divisor * increment, roundingMode);
+      seconds = rounded.divide(divisor).toJSNumber();
+      milliseconds = microseconds = nanoseconds = 0;
+      break;
+    }
+    case 'millisecond': {
+      const divisor = 1e6;
+      nanoseconds = bigInt(milliseconds).multiply(1e6).plus(bigInt(microseconds).multiply(1e3)).plus(nanoseconds);
+      total = nanoseconds.toJSNumber() / divisor;
+      const rounded = RoundNumberToIncrement(nanoseconds, divisor * increment, roundingMode);
+      milliseconds = rounded.divide(divisor).toJSNumber();
+      microseconds = nanoseconds = 0;
+      break;
+    }
+    case 'microsecond': {
+      const divisor = 1e3;
+      nanoseconds = bigInt(microseconds).multiply(1e3).plus(nanoseconds);
+      total = nanoseconds.toJSNumber() / divisor;
+      const rounded = RoundNumberToIncrement(nanoseconds, divisor * increment, roundingMode);
+      microseconds = rounded.divide(divisor).toJSNumber();
+      nanoseconds = 0;
+      break;
+    }
+    case 'nanosecond': {
+      total = nanoseconds;
+      nanoseconds = RoundNumberToIncrement(bigInt(nanoseconds), increment, roundingMode);
```

Comment:
BTW, interesting that the first parameter of this function does seem to sometimes be a number and sometimes be a bigint, unlike the second (`increment`) param that's always a Number. Good to investigate in a follow-up.

Reply:
Definitely - I think the eventual signature will be `RoundNumberToIncrement(quantity: BigInt, increment: number, roundingMode: Temporal.RoundingMode)`.

Will fix in a followup.